src/Surfnet/ServiceProviderDashboard/Domain/Service/ContractualBaseService.php

```diff
@@ -53,4 +53,16 @@ public function writeContractualBase(ManageEntity $entity): void
         // 4. Set the coin value on the entity
         $entity->getMetaData()?->getCoin()->setContractualBase($contractualBase);
     }
+
+    public function writeContractualBaseForTestEntity(ManageEntity $entity, ManageEntity $originalEntity): void
+    {
+        if ($entity->getEnvironment() !== Constants::ENVIRONMENT_TEST) {
```

Comment:
Consider using the `isProduction` method of the `ManageEntity` (or adding an `isTest` variant for this purpose.

Reply:
Changing this is very inconvenient in the test mocks, so left it as is.